File: app/src/main/java/com/asteph11/finalproject/models/data/Grid.java
package com.asteph11.finalproject.models.data;

public class Grid {

    public static final int DIMENSIONS = 7;

    public enum Status {
        EMPTY,
        SHIP,
        MISS,
        HIT
    }

    private final GridCoord[][] grid = new GridCoord[DIMENSIONS][DIMENSIONS];

    public Grid() {
        for(int i = 0; i < DIMENSIONS; i++) {
            for(int j = 0; j < DIMENSIONS; j++) {
                grid[i][j] = new GridCoord();
            }
        }
    }

    public void setStatus(int coordX, int coordY, Status status) {
        grid[coordY][coordX].setStatus(status.ordinal());
    }

    public int getStatusIndex(int coordX, int coordY) {
        return grid[coordY][coordX].getStatus();
    }

    public Status getStatusAt(int coordX, int coordY) {
        return Status.values()[getStatusIndex(coordX, coordY)];
    }

    public boolean getStatusAt(int coordX, int coordY, Status status) {
        return getStatusAt(coordX, coordY) == status;
    }

    public GridCoord[][] getGridCoords() {
        return grid;
    }

    public String toString() {
        String t = "Current grid\n";

        for(GridCoord[] gc : grid) {
            t += "| ";
            for(GridCoord g: gc) {
                t += String.format("%6s", g) + " | ";
            }
            t += "\n";
        }

        return t;
    }

    static class GridCoord {

        private int status = Grid.Status.EMPTY.ordinal();

        public void setStatus(int status) {
            this.status = status;
        }

        public int getStatus() {
            return status;
        }

        public String toString() {
            return " " + Grid.Status.values()[status];
        }

    }
}

File: app/src/main/java/com/asteph11/finalproject/models/data/Grid.kt
package com.asteph11.finalproject.models.data

import com.asteph11.finalproject.models.data.Grid.GridCoord

class Grid {
    enum class Status {
        EMPTY, SHIP, MISS, HIT
    }

    val gridCoords = Array(DIMENSIONS) { arrayOfNulls<GridCoord>(DIMENSIONS) }
    fun setStatus(coordX: Int, coordY: Int, status: Status) {
        gridCoords[coordY][coordX]!!.status = status.ordinal
    }

    fun getStatusIndex(coordX: Int, coordY: Int): Int {
        return gridCoords[coordY][coordX]!!.status
    }

    fun getStatusAt(coordX: Int, coordY: Int): Status {
        return Status.values()[getStatusIndex(coordX, coordY)]
    }

    fun getStatusAt(coordX: Int, coordY: Int, status: Status): Boolean {
        return getStatusAt(coordX, coordY) == status
    }

    override fun toString(): String {
        var t = "Current grid\n"
        for (gc in gridCoords) {
            t += "| "
            for (g in gc) {
                t += String.format("%6s", g) + " | "
            }
            t += "\n"
        }
        return t
    }

    class GridCoord {
        var status = Status.EMPTY.ordinal
        override fun toString(): String {
            return " " + Status.values()[status]
        }
    }

    companion object {
        const val DIMENSIONS = 7
        const val MAX_NUM_SHIPS = (DIMENSIONS * DIMENSIONS * (17 / 100.0)).toInt()
    }

    init {
        for (i in 0 until DIMENSIONS) {
            for (j in 0 until DIMENSIONS) {
                gridCoords[i][j] = GridCoord()
            }
        }
    }
}